test/test_jit.py

```diff
@@ -4212,20 +4212,39 @@ def test_int_to_float():
             b = float(2)
             return b + 1.0
 
-        with self.assertRaisesRegex(RuntimeError, "Cannot cast type"):
-            @torch.jit.script
-            def test_int_to_bool():
-                return bool(5)
+        @torch.jit.script
+        def test_int_to_bool(x):
+            # type: (int) -> int
+            taken = bool(1)
+            not_taken = bool(0)
+            a = x
+            if not_taken:
+                a = x + 5
+            if taken:
+                a = x + 3
+            return a
+
+        self.assertExpectedGraph(test_int_to_bool.graph, "int_to_bool")
```

Comment:
What are those expect files supposed to achieve? Maybe it would be better to test that it simply compiles and that it gives the same answers as Python functions across a range of inputs that should trigger different branches (or not)

Reply:
Refactored to have more test + no expects